.github/workflows/pull-request.yml

```diff
@@ -462,3 +462,48 @@ jobs:
         run: |
           echo "Error: Pull request target branch must be 'develop-postgres'. Please refer PR_GUIDELINES.md"
           exit 1
+
+  CodeRabbit-Approval:
+    name: Validate CodeRabbit.ai Approval
+    runs-on: ubuntu-latest
+    needs : [Code-Quality-Checks, Test-Application, Start-App-Without-Docker, Docker-Start-Check]
+    steps:
+      - name: Check CodeRabbit.ai Approval
+        run: |
+          echo "Step 1: Checking CodeRabbit.ai approval..."
+
+          echo "Fetching PR reviews from GitHub API..."
+          reviews=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/reviews")
+          
+          if [ $? -ne 0 ]; then
+            echo "Error: Failed to fetch PR reviews."
+            exit 1
+          fi
+
+          echo "Checking if the response is valid JSON..."
+          if ! echo "$reviews" | jq . >/dev/null 2>&1; then
+            echo "Error: Invalid JSON response from GitHub API."
+            echo "Response received: $reviews"
+            exit 1
+          fi
+
+          echo "Parsing the latest review by each user..."
+          latest_reviews=$(echo "$reviews" | jq -c '[.[]] | group_by(.user.login) | map(max_by(.submitted_at))')
```

Comment:
1. Please create a `.github/workflows/scripts` directory
2. Place this script there
3. Reference the script in the YAML file

It'll be easier to maintain in future

Reply:
ok sure